src/instruments/src/EFB/Ground/Ground.tsx

```diff
@@ -17,47 +17,142 @@
  */
 
 import React from 'react';
-import { IconCornerDownLeft, IconCornerDownRight, IconArrowDown, IconHandStop, IconTruck, IconBriefcase, IconBuildingBridge, IconBuildingArch, IconArchive } from '@tabler/icons'
+import { IconCornerDownLeft, IconCornerDownRight, IconArrowDown, IconHandStop, IconTruck, IconBriefcase, IconBuildingArch, IconArchive } from '@tabler/icons'
 
 import './Ground.scss'
 import fuselage from '../Assets/320neo-outline-upright.svg'
+import {setSimVar, getSimVar} from '../../util.mjs'
 
 type GroundProps = {}
 
-type GroundState = {}
+type GroundState = {
+    tugActive: boolean;
+    activeButton: string,
+}
 
 class Ground extends React.Component<GroundProps, GroundState> {
 
+    constructor(props: GroundProps) {
+        super(props);
+
+        this.state = {
+            tugActive: false,
+            activeButton: 'none',
+        };
+      }
+
+    toggleGroundAction(action: GroundServices) {
+            setSimVar(action, "1", "boolean");
+    }
+
+    setTugHeading(action: GroundServices, direction: number) {
+            if (!this.state.tugActive) {
+                this.togglePushback();
+            } else {
+                const tugHeading = this.getTugHeading(direction);
+                // KEY_TUG_HEADING is an unsigned integer, so let's convert
+                setSimVar(action, (tugHeading * 11930465) & 0xffffffff, "UINT32");
+            }
+    }
+
+    getTugHeading(value: number): number {
+        const currentHeading = getSimVar("PLANE HEADING DEGREES TRUE", "degrees");
+        return (currentHeading  + value) % 360;
+    }
+
+    togglePushback() {
+        const tugActive = this.state.tugActive;
+        setSimVar(GroundServices.TOGGLE_PUSHBACK, 1, "boolean");
+        this.setState({
+            tugActive: !tugActive
+        });
+    }
+
+    unselectButton() {
+        this.setState({
+            activeButton: "none"
+        })
+    }
+
+    handleClick(callBack: () => void, event: React.MouseEvent) {
+        this.setState({
+            activeButton: event.currentTarget.id
+        });
+        callBack();
+    }
+
     render() {
         return (
             <div className="wrapper flex-grow flex flex-col">
+                <img className="airplane w-full" src={fuselage} />
                 <div className="pushback control-grid">
                     <h1 className="text-white font-medium text-xl">Pushback</h1>
-                    <div className="stop"><IconHandStop/></div>
-                    <div className="down-left"><IconCornerDownLeft/></div>
-                    <div className="down selected"><IconArrowDown /></div>
-                    <div className="down-right"><IconCornerDownRight/></div>
+                    <div id="stop"
+                         onMouseDown={(e) => this.handleClick(() => this.state.tugActive ? this.togglePushback() : {}, e)}
+                         onMouseUp={() => this.unselectButton()}
```

Comment:
I was surprised to see the onMouseUp behaviour here. I think you're using the selected class as a pressed animation.

When I wrote the selected class I was thinking you would toggle the button's selected state, so that while you're pushing back, the back button would be highlighted so you would know what the efb is commanding the tug to do.

Reply:
Exactly. Your way makes more sense though, I'll try to adapt it this way :) If you have an idea for classes for the pressed "animation" I'd be happy to add, but I will look as well

Reply:
yeah I think a pressed animation would be good as well because it's pretty non-obvious that you've pressed a button on the efb. Whatever you think is good is ok with me